.drone.yml

```
@@ -1,297 +1,391 @@
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
+      - ENVIRONMENT=cs-qa
       - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_cs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - CS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
 
-  deploy-to-wcs-dev-from-build-number:
-    group: deploy-to-dev
+  - name: deploy to wcs-qa
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=wcs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
+      - ENVIRONMENT=wcs-qa
       - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_wcs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - WCS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
+
 
-  deployment:
+  - name: deploy to cs-dev
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=${DRONE_DEPLOY_TO}
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_dev_cs
-      - hocs_info_service_demo_cs
-      - hocs_info_service_dev_wcs
-      - hocs_info_service_demo_wcs
     commands:
       - cd kube-hocs-info-service
       - ./deploy.sh
+    environment:
+      ENVIRONMENT: cs-dev
+      KUBE_TOKEN:
+        from_secret: hocs_info_service_cs_dev
+      KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+      VERSION: build_${DRONE_BUILD_NUMBER}
     when:
-      event: deployment
-      environment: [cs-dev, cs-demo, wcs-dev, wcs-demo]
+      branch:
+        - main
+      event:
+        - push
+    depends_on:
+      - clone kube repo
 
-  deploy-to-qa-cs:
+  - name: deploy to wcs-dev
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=cs-qa
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_qa_cs
     commands:
-      - source version.txt
-      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    environment:
+      ENVIRONMENT: wcs-dev
+      KUBE_TOKEN:
+        from_secret: hocs_info-service_wcs_dev
+      VERSION: build_${DRONE_BUILD_NUMBER}
     when:
-      event: deployment
-      environment: cs-qa
+      branch:
+        - main
+      event:
+        - push
+    depends_on:
+      - clone kube repo
 
-  deploy-to-qa-wcs:
+  - name: deployment
```

As this doesn't allow prod I'd like this more exlicit. It also follows what the other service changes have been.

```suggestion
  - name: deploy to not prod
```

---

```suggestion
  - name: deployment to not-prod
```

---

Would keeping `deployment` be too long?